api/src/main/java/org/pmiops/workbench/initialcredits/InitialCreditsExpirationServiceImpl.java

```diff
@@ -129,51 +142,58 @@ public void extendInitialCreditsExpiration(DbUser user) {
   private void checkExpiration(DbUser user) {
     DbUserInitialCreditsExpiration userInitialCreditsExpiration =
         user.getUserInitialCreditsExpiration();
-    if (null != userInitialCreditsExpiration
-        && haveCreditsExpired(user)
-        && userInitialCreditsExpiration
-            .getNotificationStatus()
-            .equals(NotificationStatus.NO_NOTIFICATION_SENT)) {
-      logger.info(
-          "Initial credits expired for user {}. Expiration time: {}",
-          user.getUsername(),
-          userInitialCreditsExpiration.getExpirationTime());
-      boolean hasExhaustedWorkspace =
-          workspaceDao.findAllByCreator(user).parallelStream()
-              .anyMatch(DbWorkspace::isInitialCreditsExhausted);
-
-      // Set initial credits expired for all workspaces regardless of whether
-      // they have exhausted their credits or not.
-      workspaceDao.findAllByCreator(user).stream()
-          .filter(
-              ws ->
-                  BillingUtils.isInitialCredits(
-                      ws.getBillingAccountName(), workbenchConfigProvider.get()))
-          .filter(DbWorkspace::isActive)
-          .filter(ws -> !ws.isInitialCreditsExpired())
-          .forEach(
-              ws -> {
-                ws.setInitialCreditsExpired(true);
-                ws.setBillingStatus(BillingStatus.INACTIVE);
-                workspaceDao.save(ws);
-                deleteAppsAndRuntimesInWorkspace(ws);
-              });
-      try {
-        // If the user has already been notified about exhausting their initial credits,
-        // we do not need to notify them about expiration as well.
-        if (!hasExhaustedWorkspace) {
-          mailService.alertUserInitialCreditsExpired(user);
-          userInitialCreditsExpiration.setNotificationStatus(
-              NotificationStatus.EXPIRATION_NOTIFICATION_SENT);
-          userDao.save(user);
-        }
-
-      } catch (MessagingException e) {
-        logger.error(
-            String.format(
-                "Failed to send initial credits expiration notification for user %s",
-                user.getUserId()));
-      }
+    if (null == userInitialCreditsExpiration
+        || userInitialCreditsExpiration.getExpirationCleanupTime() != null) {
+      return;
+    }
+
+    if (haveCreditsExpired(user)) {
+      handleExpiredCredits(user, userInitialCreditsExpiration);
+    } else if (areCreditsExpiringSoon(user)
+        && null == userInitialCreditsExpiration.getApproachingExpirationNotificationTime()) {
+      handleExpiringSoonCredits(user, userInitialCreditsExpiration);
+    }
+  }
+
+  private void handleExpiredCredits(DbUser user, DbUserInitialCreditsExpiration userInitialCreditsExpiration){
```

Comment:
there's no requirement to email the user notifying them their credits expired already, correct?

Reply:
I don't see any reference to it in the doc. There is mention of getting rid of the "out credits" email, so it seems consistent to not have "expired" either.

https://docs.google.com/document/d/1h1FEFvSO48NkH6tBu5tqwPpPoE5CahGx9hZBS24ToZA/edit?tab=t.0#heading=h.1l2gwqszgnnw